components/director/internal/domain/api/fixtures_test.go

```diff
@@ -0,0 +1,344 @@
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/stretchr/testify/require"
+)
+
+func fixModelApiDefinition(id, name, description string) *model.APIDefinition {
+
+	data := []byte("")
+	spec := &model.APISpec{
+		Data: &data,
+		FetchRequest: &model.FetchRequest{
+			Mode: model.FetchModeSingle,
+			Status: &model.FetchRequestStatus{
+				Condition: model.FetchRequestStatusConditionInitial,
+			},
+		},
+	}
+	version := &model.Version{}
+
+	return &model.APIDefinition{
+		ID:            id,
+		ApplicationID: "applicationID",
+		Name:          name,
+		Description:   &description,
+		Spec:          spec,
+		Version:       version,
+	}
+}
+
+func fixGQLApiDefinition(id, name, description string) *graphql.APIDefinition {
+
+	data := graphql.CLOB("")
+	spec := &graphql.APISpec{
+		Data: &data,
+		FetchRequest: &graphql.FetchRequest{
+			Mode: graphql.FetchModeSingle,
+			Status: &graphql.FetchRequestStatus{
+				Condition: graphql.FetchRequestStatusConditionInitial,
+			},
+		},
+	}
+	version := &graphql.Version{}
+
+	return &graphql.APIDefinition{
+		ID:            id,
+		ApplicationID: "applicationID",
+		Name:          name,
+		Description:   &description,
+		Spec:          spec,
+		Version:       version,
+		Auth:          &graphql.RuntimeAuth{},
+		Auths:         []*graphql.RuntimeAuth{},
+		DefaultAuth:   &graphql.Auth{},
+	}
+}
+
+func fixModelApiDefinitionWithSpec(id, name string, spec model.APISpec) *model.APIDefinition {
+	return &model.APIDefinition{
+		ID:          id,
+		Name:        name,
+		Description: nil,
+		Spec:        &spec,
+	}
+}
+
+func fixDetailedModelApiDefinition(t *testing.T, id, name, description string, group string) *model.APIDefinition {
+
+	time, err := time.Parse(time.RFC3339, "2002-10-02T10:00:00-05:00")
+	require.NoError(t, err)
+
+	data := []byte("data")
+	filter := "filter"
+	format := model.SpecFormatJSON
+
+	spec := &model.APISpec{
+		Data:   &data,
+		Format: &format,
+		Type:   model.APISpecTypeOpenAPI,
+		FetchRequest: &model.FetchRequest{
+			URL:    "https://test-fetch-request.com",
+			Auth:   nil,
+			Mode:   model.FetchModeSingle,
+			Filter: &filter,
+			Status: &model.FetchRequestStatus{
+				Condition: model.FetchRequestStatusConditionInitial,
+				Timestamp: time,
+			},
+		},
+	}
+
+	deprecated := false
+	deprecatedSince := ""
+	forRemoval := false
+
+	version := &model.Version{
+		Value:           "1.0.0",
+		Deprecated:      &deprecated,
+		DeprecatedSince: &deprecatedSince,
+		ForRemoval:      &forRemoval,
+	}
+
+	auth1 := model.Auth{
+		Credential: model.CredentialData{
+			Basic: &model.BasicCredentialData{
+				Username: "test",
+				Password: "pwd",
+			},
+		},
+		AdditionalHeaders:     map[string][]string{"testHeader": {"hval1", "hval2"}},
+		AdditionalQueryParams: map[string][]string{"testParam": {"pval1", "pval2"}},
+		RequestAuth:           &model.CredentialRequestAuth{},
+	}
+	auth2 := model.Auth{
```

Comment:
What's the difference between auth1 and auth2? And why are you declaring it twice?

Reply:
BTW Don't pass all values for Auth, as the Auth converter is tested somewhere else. What you need to do is to test only API converter and that's it. You're mocking Auth converter anyway, so please decrease complexity here.